pydis_site/apps/resources/templatetags/get_category_icon.py

```diff
@@ -0,0 +1,38 @@
+from django import template
+
+register = template.Library()
+
+
+@register.filter
+def get_category_icon(name: str) -> str:
+    """Get icon of a specific resource category."""
+    icons = {
+        "Algorithms And Data Structures": "fa-cogs",
+        "Data Science": "fa-flask",
+        "Databases": "fa-server",
+        "Game Development": "fa-joystick",
+        "General": "fa-book",
+        "Microcontrollers": "fa-microchip",
+        "Other": "fa-question-circle",
+        "Software Design": "fa-paint-brush",
+        "Testing": "fa-vial",
+        "Tooling": "fa-toolbox",
+        "User Interface": "fa-desktop",
+        "Web Development": "fa-wifi",
+        "Discord Bots": "fa-robot",
+        "Book": "fa-book",
+        "Community": "fa-users",
+        "Course": "fa-chalkboard-teacher",
+        "Interactive": "fa-mouse-pointer",
+        "Podcast": "fa-microphone-alt",
+        "Tool": "fa-tools",
+        "Tutorial": "fa-clipboard-list",
+        "Video": "fa-video",
+        "Free": "fa-first-aid",
+        "Paid": "fa-sack",
+        "Subscription": "fa-credit-card",
+        "Beginner": "fa-play-circle",
+        "Intermediate": "fa-align-center"
+    }
+    icon_name = icons[name]
```

Comment:
This dictionary should be a global so that it's not being re-created each time. Try naming it `_ICONS`.

Reply:
Pretty sure Python handles this and uses LOAD CONST, that said using an all-caps variable at the top of the file is cleaner.

Reply:
Also I wish we had function composition so that this function could just be `get_category_icon = register.filter @ 'fa {icon_name}'.format @ icons.get`, but no :disappointed: